annotations/servicebinding-annotations/src/main/java/io/dekorate/servicebinding/annotation/Application.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2018 The original authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.dekorate.servicebinding.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import io.dekorate.kubernetes.config.Configuration;
+import io.sundr.builder.annotations.Buildable;
+import io.sundr.builder.annotations.Pojo;
+
+@Buildable(builderPackage = "io.fabric8.kubernetes.api.builder")
+@Pojo(name = "ApplicationConfig", mutable = true, superClass = Configuration.class, relativePath = "../config", withStaticAdapterMethod = false)
+@Target({ ElementType.CONSTRUCTOR, ElementType.TYPE })
+@Retention(RetentionPolicy.SOURCE)
+public @interface Application {
+
+  String group() default "";
```

Comment:
`@Application` annotation is optional, the developers need not specify those. I too thought of having `BindingPath` directly under `@ServiceBinding`  but then I thought that would deviate the actual servicebinding CR yaml structure and would make it difficult for users to understand. ServiceBinding CR yaml is not too complex, so I thought I will just follow it and make it easy for developers to write the annotations keeping the yaml structure in mind.

Based on your experience writing dekorate annotations, do you think it's a good idea to deviate from the original yaml structure even if it is not too complex? if so I can bring it under `@ServiceBinding` directly.

Reply:
So far, we deviated from the original structure either for techincal reasons, or to keep things simpler. If the use of the annotation is optional, then I don't have a very strong opinion. 
If you feel that it will make it easier for the users to understand, we can pick it and just explain in javadoc, how these values will be poplulated when the annotation is not used.

Reply:
sometimes I feel it's okay to deviate from the original structure, the reason why I think it's okay is almost 99.99% of the times we don't need to use `@Application` annotation unless you are specifying the `@BindingPath` annotation in the `@Application` because all of the other parameters can be populated by us, if we bring `@BindingPath` directly under `@ServiceBinding` , developers don't have to do this ```application = @Application(bindingPath = @BindingPath(secretPath = "xyz"))``` and need to do this ```bindingPath = @BindingPath(secretPath = "xyz")``` under `@ServiceBinding` which makes it simple

I will move `@BindingPath` directly under `@ServiceBinding`